### PR TITLE
ISSUE-618: recreate the client each time is_router_connected fails

### PR DESCRIPTION
**Pull request by kgiusti**

### CODE REVIEW

File: tests/system_test.py [hunk 1]
```diff
@@ -846,19 +846,23 @@ def wait_ready(self, **retry_kwargs):
...
+
+        # TODO aconway 2015-01-29: The above check should be enough, we
+        # should not advertise a remote router in management till it is fully
+        # connected. However we still get a race where the router is not
+        # actually ready for traffic. Investigate.
+        # Meantime the following actually tests send-thru to the router.
+        try:
+            with Node.connect(self.addresses[0], router_id) as node:
+                return node.query('io.skupper.router.router')
+        except Exception as exc:
```
Comment (jiridanek) [inline tests/system_test.py:863]: I don't like this broad except. It catches syntax errors and so on. That makes it hard to revisit such test code in the future.

Comment (jiridanek) [reply tests/system_test.py:863]: I'd put back the line

```
except (proton.ConnectionException, NotFoundStatus, proton.utils.LinkDetached) as exc:
```

and probably the associated comment?

**Review by jiridanek — APPROVED** [review]
Looks ok, restoring the aconway comment is good, but I am not fan of that `except Exception` block.

**Comment by jiridanek** [comment]
This might maybe help with https://github.com/skupperproject/skupper-router/issues/126, I don't remember the details there, but recreating the connection for every check should avoid the possibility of (one of the) failures in that ticket.

File: tests/system_test.py (hunk 1 above)
Comment (kgiusti) [reply tests/system_test.py]: Ok, I've restored it.

**Comment by jiridanek** [comment]
Thanks a lot for adding the specific exceptions. I hope it will be good for something in the future. More reliable tests!?

**Pull request merged**